scheduler/reconcile_util.go

```diff
@@ -209,15 +224,22 @@ func (a allocSet) fromKeys(keys ...[]string) allocSet {
 	return from
 }
 
-// filterByTainted takes a set of tainted nodes and filters the allocation set
-// into three groups:
+// TODO: This might need to be a new separate method to avoid impacting the
+// system scheduler OR the system scheduler may actually require updates.
+// groupByAllocOrNodeStatus takes a set of tainted nodes and filters the allocation set
+// into 5 groups:
 // 1. Those that exist on untainted nodes
 // 2. Those exist on nodes that are draining
 // 3. Those that exist on lost nodes
-func (a allocSet) filterByTainted(nodes map[string]*structs.Node) (untainted, migrate, lost allocSet) {
+// 4. Those that are on nodes that are disconnected, but have not had their ClientState set to unknown
+// 5. Those that have had their ClientState set to unknown, but their node has reconnected.
+func (a allocSet) groupByAllocOrNodeStatus(taintedNodes map[string]*structs.Node) (untainted, migrate, lost, disconnecting, reconnecting allocSet) {
```

Comment:
#Refactoring review

Reply:
This method is a filter-then-map, which I think may be why both this rename and the original name are lacking. And `filterByTaintedAndGroupByStatus` is veering close to awful Java-style names. But I feel like the filter is the important and non-obvious part of this name; the function type signature already tells you that it's grouping.